src/OrchardCore.Modules/OrchardCore.Lucene/Recipes/LuceneIndexResetStep.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Threading.Tasks;
+using OrchardCore.BackgroundJobs;
+using OrchardCore.Recipes.Models;
+using OrchardCore.Recipes.Services;
+
+namespace OrchardCore.Lucene.Recipes
+{
+    /// <summary>
+    /// This recipe step resets a lucene index.
+    /// </summary>
+    public class LuceneIndexResetStep : IRecipeStepHandler
+    {
+        private readonly LuceneIndexSettingsService _luceneIndexSettingsService;
+        private readonly LuceneIndexingService _luceneIndexingService;
+        private readonly LuceneIndexManager _luceneIndexManager;
+
+        public LuceneIndexResetStep(
+            LuceneIndexSettingsService luceneIndexSettingsService,
+            LuceneIndexingService luceneIndexingService,
+            LuceneIndexManager luceneIndexManager)
+        {
+            _luceneIndexSettingsService = luceneIndexSettingsService;
+            _luceneIndexManager = luceneIndexManager;
+            _luceneIndexingService = luceneIndexingService;
+        }
+
+        private async Task ResetIndexAsync(string indexName)
```

Comment:
This method will no longer be necessary. Look at my Elasticsearch PR and copy what I did.

Reply:
I may have made a copy/paste mistake. But they essentially are quite the same. Except rebuild will drop the index entirely and recreate it.